code/datums/storage/storage.dm

```diff
@@ -347,6 +347,10 @@ GLOBAL_LIST_EMPTY(cached_storage_typecaches)
 	if(QDELETED(to_insert) || !istype(to_insert))
 		return FALSE
 
+	//stops you from putting stuff like off-hand thingy inside. Hologram storages can accept only hologram items
+	if((to_insert.item_flags & ABSTRACT) || ((parent.flags_1 & HOLOGRAM_1) && !(to_insert.flags_1 & HOLOGRAM_1)))
+		return FALSE
```

Comment:
it's a lot more readable if we split it across two, i hate how all these operators look
```suggestion
	if(to_insert.item_flags & ABSTRACT)
		return FALSE
	if(parent.flags_1 & HOLOGRAM_1) && !(to_insert.flags_1 & HOLOGRAM_1)
		return FALSE
```

Reply:
also re-reading it now that it's split - you don't seem to do what you did in the original pr (preventing holographic items from being inserted in non-holographic storage). intended?

Reply:
Yeah you right. I burnt myself here.

Can you rewrite your suggestion like this.

```
if(parent.flags_1 & HOLOGRAM_1) //hologram storage
  if(!(to_insert.flags_1 & HOLOGRAM_1)) //no real items
    return
else if(to_insert.flags_1 & HOLOGRAM_1)//real storage no holograms 
  return
```

So I can commit it

Reply:
Personally I just write the suggestion myself but ok 
```suggestion
	if(to_insert.item_flags & ABSTRACT)
		return FALSE
	if(parent.flags_1 & HOLOGRAM_1)
		if(!(to_insert.flags_1 & HOLOGRAM_1))
			return FALSE
	else if(to_insert.flags_1 & HOLOGRAM_1)
		return FALSE
```

Reply:
I'm on mobile and you know can't type tabs. Thanks